modules/renter/contractor/contractor.go

```diff
@@ -85,14 +85,12 @@ func (c *Contractor) Contract(hostAddr modules.NetAddress) (modules.RenterContra
 	return modules.RenterContract{}, false
 }
 
-// Contracts returns the contracts formed by the contractor.
+// Contracts returns the contracts formed by the contractor. Only contracts
+// formed with currently online hosts are returned.
 func (c *Contractor) Contracts() (cs []modules.RenterContract) {
 	c.mu.RLock()
 	defer c.mu.RUnlock()
-	for _, c := range c.contracts {
-		cs = append(cs, c)
-	}
-	return
+	return c.onlineContracts()
```

Comment:
>lowercase functions assume that they are being called with an exclusive lock

what? I don't recall any such convention. It's certainly not in our `Developers.md`. You wrap unexported functions with readlocks in both the (old) host and consensus set. When are you supposed to use readlocks if you can't call unexported functions with them?